nixos/modules/services/hardware/irexec.nix

```
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.irexec;
+in {
+
+  ###### interface
+
+  options = {
+    services.irexec = {
```

Maybe you could remove the extra layer of nesting?

```suggestion
  options.services.irexec = {
```